src/secondary.jl

```diff
@@ -0,0 +1,166 @@
+export run_dssp, run_stride, ss_code!, ss_code, helixselector, sheetselector, coilselector
+
+dssp_executable = `$(DSSP_jll.mkdssp()) --mmcif-dictionary $(DSSP_jll.mmcif_pdbx_dic)`
+const dssp_init_line = "  #  RESIDUE AA STRUCTURE BP1 BP2  ACC     N-H-->O    O-->H-N    N-H-->O    O-->H-N    TCO  KAPPA ALPHA  PHI   PSI    X-CA   Y-CA   Z-CA"
+
+stride_executable = STRIDE_jll.stride_exe()
+
+# Write a PDB file with header and the ATOM records for DSSP/STRIDE to read
+function writesubpdb(filepath::AbstractString, el::ProteinStructure)
+    open(filepath, "w") do io
+        print(
+            io,
+            "HEADER    XXXXXXXXXXXXXXXXXXXXXX                  25-DEC-23   XXXX              ",
+        )
+        for at in collectatoms(el)
+            checkchainerror(at)
+            println(io, pdbline(at))
+        end
+    end
+end
+
+function run_dssp(curr_model::Model, mod_n::Int)
+    struc = ProteinStructure()
+    struc.models[mod_n] = curr_model
+    fixlists!(struc)
+
+    # Write the structure to a temporary PDB file
+    tmp_pdb_path = tempname() * ".pdb"
+    writesubpdb(tmp_pdb_path, struc)
+
+    # Run DSSP on the temporary PDB file
+    dssp_output = try
+        readchomp(pipeline(`$dssp_executable --output-format dssp $tmp_pdb_path`))
+    catch
+        "error"
```

Comment:
Probably better to let this error loudly to the user, the error could be informative.